### PR TITLE
Enable a tracer defined in the servlet context to override a tracer a…

### PR DESCRIPTION
**Pull request by objectiser**
…ssociated with the TraceFilter

### CODE REVIEW

File: opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/TracingFilter.java [hunk 1]
```diff
@@ -109,6 +109,15 @@ public void init(FilterConfig filterConfig) throws ServletException {
         this.filterConfig = filterConfig;
         ServletContext servletContext = filterConfig.getServletContext();
 
+        // Check whether the servlet context provides a tracer
+        Tracer scTracer = (Tracer)servletContext.getAttribute(Tracer.class.getName());
```
Comment (jpkrohling) [inline opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/TracingFilter.java]: A safer approach is to get an Object and check if it's an instance of Tracer before casting.

```java
Object tracerObj = servletContext.getAttribute(Tracer.class.getName());
if (tracerObj instanceof Tracer) {
  // no need for the null check!
  tracer = (Tracer) tracerObj;
}
```

**Review by jpkrohling — APPROVED** [review]
A minor comment on the cast to `Tracer`, but LGTM.

**Comment by pavolloffay** [comment]
It introduces yet another way how the tracer can be provided which can cause some confusion.

Why is not `GlobalTracer` sufficient?

**Comment by objectiser** [comment]
If the OT dependencies are installed in the servlet container, then the `GlobalTracer` is global to the container - so only one tracer - which also means all webapps reporting spans for the same service name.

This PR enables the tracer to be managed with the servlet context lifecycle.

**Comment by pavolloffay** [comment]
Could you please add this to readme and javadoc? Explaining when it should be used.

**Comment by objectiser** [comment]
@pavolloffay done

**Review by pavolloffay — APPROVED** [review]

**Comment by objectiser** [comment]
@pavolloffay @jpkrohling thanks.

**Pull request merged**